libraries/resource_providers/api200/logical_interconnect_provider.rb

```diff
@@ -82,17 +82,22 @@ def firmware_handler(action)
         end
       end
 
-      def load_port(ports)
-        port = @item.get_unassigned_uplink_ports_for_port_monitor.select { |k| k['portName'] == ports['analyzerPort']['port_name'] }.first
-        raise("Port '#{ports['analyzerPort']['port_name']}' was not found or is already being monitored!") unless port
-        load_downlink_ports(ports['monitoredPorts'], port['interconnectName'])
-        ports['analyzerPort']['portUri'] = port['uri']
-        ports['analyzerPort'].delete('port_name')
-        ports['monitoredPorts'].each { |k| k.delete('port_name') }
+      def load_ports
+        @item.retrieve! || raise("#{@resource_name} '#{@name}' not found!")
```

Comment:
I thought of doing it that way. Then I realized that I would have to manipulate another object, like ```temp = key ? { key.to_s => Marshal.load(Marshal.dump(@item[key])) } : Marshal.load(Marshal.dump(@item.data))``` and it would practically be difficult to reuse the ```update_handler``` method.

Reply:
Not sure I understand. 

Why would you need to do all that to run `@item.retrieve! || raise("#{@resource_name} '#{@name}' not found!")` in `update_port_monitor` instead of here?

If you just move this from [line 86](https://github.com/HewlettPackard/oneview-chef/pull/325/files/cb5f7cc7d99d73a925bedd963b466960cf9e7e00..c59a4c4e2bf7d9b21c1582ff5394903025db65e6#diff-4b902556c979d8bc0791ddf2cac48843R86) to [line 157](https://github.com/HewlettPackard/oneview-chef/pull/325/files/cb5f7cc7d99d73a925bedd963b466960cf9e7e00..c59a4c4e2bf7d9b21c1582ff5394903025db65e6#diff-4b902556c979d8bc0791ddf2cac48843R157) it should work just fine unless I'm missing something.

Reply:
Sorry @fgbulsoni , I was not clear. When I run an ```@item.retrieve!```. the data is replaced. Being necessary to store in another variable, as is done in the ```update_handler```.

Reply:
Ok, so, both of that load/unload and the retrieve likely belong in the `update_port_monitor` method, I'm guessing in that order. Right?

Reply:
Yeah! I divided the algorithm into more than one method to split the tasks.